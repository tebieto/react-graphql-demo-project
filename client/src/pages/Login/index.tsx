import React from 'react';
import CustomButton from '../../components/CustomButton';
import CustomInput from '../../components/CustomInput';
import Title from '../../components/Title';
import { getPageTitle } from '../../utils';
import { LoginContainer } from './styles';

const Login = (): JSX.Element => {
  const handleSubmit = React.useCallback(() => {
    alert('here');
  }, []);

  const handleChange = React.useCallback(() => {
    alert('here');
  }, []);
  return (
    <LoginContainer>
      <Title title={getPageTitle('Login')} />
      <CustomInput onChange={handleChange} placeholder="Enter Email" />
      <CustomInput
        onChange={handleChange}
        type="password"
        placeholder="Enter Password"
      />
      <CustomButton onClick={handleSubmit}>Login</CustomButton>
    </LoginContainer>
  );
};

export default Login;
