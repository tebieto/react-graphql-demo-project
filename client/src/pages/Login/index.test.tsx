import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Login from '.';

it('renders correctly', () => {
  const tree = mount(<Login />);
  expect(toJson(tree)).toMatchSnapshot();
  expect(tree.contains('Login to Github')).toEqual(true);
});
