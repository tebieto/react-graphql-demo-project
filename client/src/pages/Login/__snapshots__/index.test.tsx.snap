// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Login>
  <styled.div>
    <div
      className="sc-gtsrHT cAgKLY"
    >
      <Title
        title="Github Search | Login"
      />
      <GithubAuth>
        <styled.div>
          <div
            className="sc-bdnxRM fqXRSa"
          >
            <u
              buttonText="Login to Github"
              className="github-login-button"
              clientId="4f262cc9e20d3043da02"
              onFailure={[Function]}
              onRequest={[Function]}
              onSuccess={[Function]}
              popupHeight={650}
              popupWidth={500}
              redirectUri=""
              scope="user:email"
            >
              <button
                className="github-login-button"
                disabled={false}
                onClick={[Function]}
              >
                Login to Github
              </button>
            </u>
          </div>
        </styled.div>
      </GithubAuth>
    </div>
  </styled.div>
</Login>
`;
